src/main/scala/translating/BAPLoader.scala

```diff
@@ -124,13 +128,11 @@ object BAPLoader {
     }
     val line = visitQuoteString(ctx.tid.name)
     val insn = parseFromAttrs(ctx.attrs, "insn").getOrElse("")
-    BAPDirectCall(
-      parseAllowed(visitQuoteString(ctx.callee.tid.name).stripPrefix("@")),
-      visitExp(ctx.cond),
-      returnTarget,
-      line,
-      insn
-    )
+    val function = visitQuoteString(ctx.callee.tid.name).stripPrefix("@")
+    if (knownNonReturningFunctions.contains(function))
+      BAPDirectCall(parseAllowed(function), visitExp(ctx.cond), None, line, insn)
+    else
+      BAPDirectCall(parseAllowed(function), visitExp(ctx.cond), returnTarget, line, insn)
```

Comment:
As I've said, I would really like to keep BAPLoader focused on representing what we get from BAP and directly translating that to the IR. Identifying non-returning calls like this from the function names should happen later on and modify the Program, not happen during the parsing & loading. We want to try to keep the steps separated.

Reply:
It was more efficient to do it that way, but have moved it to nonReturningFunction now